src/app/ipv/middleware.js

```diff
@@ -64,8 +64,13 @@ async function fetchUserDetails(req) {
   return generateUserDetails(userDetailsResponse, req.i18n);
 }
 
-async function handleJourneyResponse(req, res, action) {
+async function handleJourneyResponse(req, res, action, currentPage = "") {
+  if (currentPage) {
+    action += `?currentPage=${currentPage}`;
```

Comment:
Matter of opinion, but I think I'd slightly prefer `currentPageId` or just `pageId`? Feels more precise about what exactly the value is.

Reply:
(I actually meant for the actual query param, rather than just this variable, but I don't feel strongly if you want to avoid making the change all the way through the stack)